src/swagger/specGenerator.ts

```diff
@@ -6,6 +6,253 @@ import { Swagger } from './swagger';
 export class SpecGenerator {
   constructor(protected readonly metadata: Tsoa.Metadata, protected readonly config: SwaggerConfig) { }
 
+  public GetSpec() {
+    let spec: Swagger.Spec = {
```

Comment:
@WoH would you be willing to review this PR? I think you’re most familiar with the spec generator code at this point.

Reply:
@aleangel1212 I noticed that a lot of this code is a copy and paste of the specGenerator 2 code. I think there has to be a DRYer way to do this. And also, shouldn’t the custom attributes get different validations depending on if the consumer has selected Swagger 2 vs OpenAPI 3.0? I ask because I believe  (and @endor could confirm) that the SpecGenerator2 and SpecGenerator3 files exist for that very reason.

Reply:
This almost positive this code was introduced due to a faulty merge. I'd take a closer look at the whole PR if there is anyone ready to pick it up.